### PR TITLE
feat: Create a SentryException from a NSException

### PR DESCRIPTION
**Pull request by bruno-garcia**
## :scroll: Description

Creates a SentryException from an `NSException`.

## :bulb: Motivation and Context
`captureException` is expected to create an event of type `error` but because we use only `message` field of the event, Sentry has no idea it was an error.

## :green_heart: How did you test it?
Unit tests.
I raised an event and checked `type=error`. The same event before the change it was `type=default`.

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [x] I updated the CHANGELOG
- [ ] I updated the docs if needed
- [ ] No breaking changes

## :crystal_ball: Next steps

This will fit well on 6.0.0 since it **could change** grouping. Needs to check with the ingestion folks.

### CODE REVIEW

**Comment by bruno-garcia** [comment]
We should add a bot that pushes the linting fixes to the PR like `getsentry/sentry` has.
A broken build due to linting is such an annoying waste of time

File: Sources/Sentry/SentryClient.m [hunk 1]
```diff
@@ -128,7 +129,10 @@ - (SentryId *)captureException:(NSException *)exception
 - (SentryEvent *)buildExceptionEvent:(NSException *)exception
 {
     SentryEvent *event = [[SentryEvent alloc] initWithLevel:kSentryLevelError];
-    event.message = exception.reason;
+    SentryException *sentryException =
+        [[SentryException alloc] initWithValue:[NSString stringWithFormat:@"%@", exception.reason]
+                                          type:exception.name];
```
Comment (philipphofmann) [inline Sources/Sentry/SentryClient.m:134]: `l`: I think we don't need the `NSString stringWithFormat` in here.
```suggestion
    SentryException *sentryException = [[SentryException alloc] initWithValue:exception.reason type:exception.name];
```

**Review by philipphofmann — COMMENTED** [review]
The changes look good to me. I just remember @HazAT mentioning something about not making use of the exceptions on the event for some reason, but I think this was about the stacktrace. @HazAT can you have a quick look if you have reasons against this PR?

**Comment by philipphofmann** [comment]
This has an impact on the grouping of the events. We should include this also in 6.0.0 or then in 6.1.0, but we should ship it together with making use of `NSError.code` and `NSError.domain` in `captureError` instead of `localizedDescription`.

**Comment by bruno-garcia** [comment]
Don't believe it affect grouping since it doesn't affect the stack traces, only the message.
@mitsuhiko can comment?

**Comment by mitsuhiko** [comment]
Exception type is also considered. Best to check directly by ingesting the event.

**Comment by bruno-garcia** [comment]
In any regard I just realize we don't have a stacktrace here, just the message so it would for sure mess with grouping.

**Review by HazAT — APPROVED** [review]
💚

**Pull request merged**